pkg/api-server/filters/filtering.go

```diff
@@ -13,6 +13,47 @@ import (
 	"github.com/kumahq/kuma/pkg/core/validators"
 )
 
+// Resource return a store filter depending on the resource. We take a descriptor so that we can do advance filtering options
+// For example we could make a filter that works on top level targetRef by looking at the descriptor info
+func Resource(resDescriptor core_model.ResourceTypeDescriptor) func(request *restful.Request) (store.ListFilterFunc, error) {
+	switch resDescriptor.Name {
+	case mesh.DataplaneType:
+		return func(request *restful.Request) (store.ListFilterFunc, error) {
```

Comment:
nit: I'd extract all cases bodies to functions. `func DataplaneFilter((request *restful.Request) (store.ListFilterFunc, error) {`, `ExternalServicesFilter` etc.

Reply:
I've looked into it and the actual refactor makes things harder to read IMO